src/app/services/coffeeShops.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
+import { catchError, Observable, throwError } from 'rxjs';
+import { NGXLogger } from 'ngx-logger';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoffeeShopsService {
+
+  constructor(private http: HttpClient,
+              private ngxLogger: NGXLogger) {
+  }
+
+  getToken(): Observable<any> {
+    return this.http.post(`https://blue-bottle-api-test.herokuapp.com/v1/tokens`, {});
+  }
```

Comment:
Doesn't it seem weird for the auth token to be obtained using the `coffeeShops` service? It is sending requests to a different endpoint after all. How would you refactor this?

Reply:
We can create an aditional service named Auth, and create there the endpoint for the token.